codegen/function_calls.java

```diff
@@ -43,12 +43,12 @@ public int func_int(int i) {
   }
   public int[] func_int_array(int[] arr) { return arr; }
   public boolean func_boolean(boolean b) { return b; }
-  public boolean[] func_boolean_array(boolean[] arr) { return arr; }
+  public int[] func_boolean_array(int[] arr) { return arr; }
   public int decrease(int i) {
     i = i - 1;
     return i;
   }
-  public int func(int i, int[] int_arr, boolean b, boolean[] b_arr, B c_b) {
+  public int func(int i, int[] int_arr, boolean b, int[] b_arr, B c_b) {
```

Comment:
`b_arr` is used further down in an `if`. It has to be replaced. I'd say remove `b_arr` completely from this function.